v2/i18n/bundle.go

```diff
@@ -26,9 +26,59 @@ type Bundle struct {
 	matcher          language.Matcher
 }
 
+// The matcher in x/text/language does not handle artificial languages,
+// see https://github.com/golang/go/issues/45749
+// This is a simplified matcher that delegates to the x/text/language matcher for
+// the harder cases.
+type matcher struct {
+	tags           []language.Tag
+	defaultMatcher language.Matcher
+}
+
+func newMatcher(tags []language.Tag) language.Matcher {
+	var hasArt bool
+	for _, tag := range tags {
+		base, _ := tag.Base()
+		hasArt = base == artTagBase
+		if hasArt {
+			break
+		}
+	}
+
+	if !hasArt {
+		return language.NewMatcher(tags)
+	}
+
+	return matcher{
+		tags:           tags,
+		defaultMatcher: language.NewMatcher(tags),
+	}
+}
+
+func (m matcher) Match(t ...language.Tag) (language.Tag, int, language.Confidence) {
+	for _, candidate := range t {
+		base, _ := candidate.Base()
+
+		if base != artTagBase {
+			break
```

Comment:
Are you sure this will actually solve the problem your upstream users are facing? This logic here bails as soon as it sees a non-artificial language, so this workaround will only work if the artificial language is first in the list. Is that always the case with Hugo?

Reply:
Hugo creates one Localizer per language, so there will always be only 1 language in that list. But you are right. I added that break as a premature optimization, I will fix it. Thanks.